qa/rpc-tests/bip135-genvbvoting-forks.py

```diff
@@ -0,0 +1,680 @@
+        self.tip = block.sha256
+        self.height += 1
+        yield TestInstance([[block, True]])
+
+        assert_equal(self.get_bip135_status(bipName)['status'], 'active')
+        tmpl = self.nodes[0].getblocktemplate({})
+        assert(bipName in tmpl['rules'])
+        assert(bipName not in tmpl['vbavailable'])
+        assert_equal(tmpl['vbrequired'], 0)
+        assert(not (tmpl['version'] & (1 << bitno)))
+
+        # Test 6
+        # Check that the new sequence lock rules are enforced
+        spendtx = self.create_transaction(self.nodes[0],
+                self.coinbase_blocks[1], self.nodeaddress, 1.0)
+        invalidate(spendtx)
+        spendtx = self.sign_transaction(self.nodes[0], spendtx)
+        spendtx.rehash()
+        invalidatePostSignature(spendtx)
+        spendtx.rehash()
+
+        block = create_block(self.tip, create_coinbase(self.height), self.last_block_time + 1)
+        block.nVersion = 5
+        block.vtx.append(spendtx)
+        block.hashMerkleRoot = block.calc_merkle_root()
+        block.rehash()
+        block.solve()
+        self.last_block_time += 1
+        yield TestInstance([[block, False]])
+        # Restart all
+        stop_nodes(self.nodes)
+        wait_bitcoinds()
+        shutil.rmtree(self.options.tmpdir)
+        self.setup_chain()
+        self.setup_network()
+        self.test.clear_all_connections()
+        self.test.add_all_connections(self.nodes)
+        NetworkThread().start() # Start up network handling in another thread
+
+    def test_BIP135GraceConditions(self):
+
+        # the fork bits used to check grace period conditions
+        gracebits = self.defined_forks[7:22]
+
+        print("begin test_BIP135GraceConditions test")
+        node = self.nodes[0]
+        self.tip = int("0x" + node.getbestblockhash(), 0)
+        header = node.getblockheader("0x%x" % self.tip)
+        assert_equal(header['height'], 0)
+
+        # Test 1
+        # generate a block, seems necessary to get RPC working
+        self.coinbase_blocks = self.nodes[0].generate(1)
+        self.height = 2
+        self.last_block_time = int(time.time())
+        self.tip = int("0x" + node.getbestblockhash(), 0)
+        test_blocks = self.generate_blocks(1, 0x20000000)  # do not set bit 0 yet
+        yield TestInstance(test_blocks, sync_every_block=False)
+
+        bcinfo = self.nodes[0].getblockchaininfo()
+        # check bits 7-15 , they should be in DEFINED
+        for f in gracebits:
+            assert_equal(bcinfo['bip135_forks'][f]['bit'], int(f[10:]))
+            assert_equal(bcinfo['bip135_forks'][f]['status'], 'defined')
+
+        # move to starttime
+        moved_to_started = False
+        while self.last_block_time < self.fork_starttime or self.height % 10:
+            test_blocks = self.generate_blocks(1, 0x20000000)
+            yield TestInstance(test_blocks, sync_every_block=False)
+            time.sleep(3)  # need to actually give daemon a little time to change the state
```

Comment:
After splitting up the test into two separate ones (as I wanted to anyway), I removed the sleeps and indeed it worked without them. Will remove.